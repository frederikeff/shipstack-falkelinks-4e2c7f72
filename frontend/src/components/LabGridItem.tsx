"use client";
import Link from "next/link";
import { trackEvent } from "@/lib/analytics";

interface LabGridItemProps {
  href: string;
  title: string;
}

export default function LabGridItem({ href, title }: LabGridItemProps) {
  return (
    <Link
      href={href}
      className="flex h-32 w-32 items-center justify-center rounded-lg bg-pink-500 p-4 text-center font-bold text-white shadow-lg transition-transform hover:scale-105"
      onClick={() => trackEvent('Lab Grid Item Click', { href, title })}
    >
      {title}
    </Link>
  );
}