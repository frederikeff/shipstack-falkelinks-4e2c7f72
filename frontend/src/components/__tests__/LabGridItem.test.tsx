import { render, screen } from '@testing-library/react'
import LabGridItem from '../LabGridItem'

it('renders correctly', () => {
  const { container } = render(
    <LabGridItem href="https://example.com" title="Example" />
  )
  expect(container).toMatchSnapshot()
})