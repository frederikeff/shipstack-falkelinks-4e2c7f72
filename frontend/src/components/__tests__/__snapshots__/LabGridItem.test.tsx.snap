// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders correctly 1`] = `
<div>
  <a
    class="flex h-32 w-32 items-center justify-center rounded-lg bg-brand-pink p-4 text-center font-bold text-white shadow-lg transition-transform hover:scale-105"
    href="https://example.com"
  >
    Example
  </a>
</div>
`;
